manim/mobject/svg/text_mobject.py

```diff
@@ -1387,3 +1387,75 @@ def extract_color_tags(self):
 
     def __repr__(self):
         return f"MarkupText({repr(self.original_text)})"
+
+
+@contextmanager
+def register_font(font_file: typing.Union[str, Path]):
+    """Context Manager which adds a font
+    file to Pango's search path.
+    This searches for the font_file at various places.
+
+    The order it searches it described below.
+
+    1. Absolute path.
+    2. In ``assets/fonts`` folder.
+    3. In ``font/`` folder.
+    4. In the same directory.
+
+    .. note ::
+
+        This method of adding font files also works with
+        :class:`CairoText`
+
+    .. important ::
+
+        This method isn't available for macOS. Using this
+        method on macOS will raise an :class:`AttributeError`.
+
+    Parameters
+    ----------
+    font_file :
+        The font file to add.
+
+    Examples
+    --------
+    Use ``with register_font(...)`` to add a font file to search
+    path.
+
+    .. code-block:: python
+
+        with register_font("path/to/font_file.ttf"):
+           a=Text("Hello",font="Custom Font Name")
+
+    Raises
+    ------
+    FileNotFoundError:
+        If the font doesn't exists.
+    """
+
+    input_folder = Path(config.input_file).parent.resolve()
+    possible_paths = [
+        Path(font_file),
+        input_folder / "assets/fonts" / font_file,
+        input_folder / "fonts" / font_file,
+        input_folder / font_file,
+    ]
+    for path in possible_paths:
+        path = path.resolve()
+        print(path)
+        if path.exists():
+            file_path = path
+            logger.debug("Found file at %s", file_path.absolute())
+            break
+    else:
+        error = f"Can't find {font_file}." f"Tried these : {possible_paths}"
+        raise FileNotFoundError(error)
+
+    try:
+        assert manimpango.register_font(str(file_path))
+        yield
+    finally:
+        if sys.platform.startswith("linux"):
+            manimpango.unregister_font()
+        else:
+            manimpango.unregister_font(str(file_path))
```

Comment:
Nitpick (non-blocking): It sounds like this check could be made inside `unregister_font` instead, no?

Reply:
Well yeah, this can be done in manimpango itself, but currently, this is how the API is.

Reply:
@leotrs can you make an issue in the other repo so that it can be fixed?

Reply:
Done